### PR TITLE
Start work on checking of too broad local imports outside of unit tests

### PR DESCRIPTION
**Pull request by JackStouffer**
Let's see where this breaks, and then I'll fix them

### CODE REVIEW

**Comment by JackStouffer** [comment]
ping @wilzbach, thoughts

**Comment by WalterBright** [comment]
Nice work, keep it coming!

**Comment by WalterBright** [comment]
BTW, let this pass the autotester, so I can pull it. Then open a new PR to continue. That way things don't get too overly large and complex.

**Comment by JackStouffer** [comment]
@WalterBright Ok, I'll stop here for now.

**Comment by wilzbach** [comment]
> ping @wilzbach, thoughts

Not really, but imho we should build specifying detailed imports into the compiler (or build a library on top of it), until then doing it manually seems a bit laborious. One of my goals for the near future is to enable more dscanner flags (one important one is to warn against undocumented public symbols), but it got a bit stalled after I realized that it's quite a workload due to so many accidentally exposed symbols. Maybe that would be a good target for the next monthly "call for action" sprint ;-)

File: std/net/isemail.d [hunk 1]
```diff
@@ -1752,12 +1752,12 @@ enum AsciiToken
  */
 T max (T) (T[] arr)
 {
-    import std.algorithm/* : max*/;
+    static import std.algorithm.comparison;
```
Comment (wilzbach) [inline std/net/isemail.d]: how about `import std.algorithm.comparison : maxFun = max;`?

Comment (JackStouffer) [inline std/net/isemail.d]: My goal for this PR is to only change the import code. There's a lot in these modules that needs to be fixed; I saw at least 15 package wide std.algorithm imports.

**Comment by WalterBright** [comment]
Auto-merge toggled off

**Comment by WalterBright** [comment]
Turned off automerge due to test failures. Needs more work!

**Comment by JackStouffer** [comment]
@WalterBright Fixed

**Comment by JackStouffer** [comment]
@WalterBright Green all around

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Pull request merged**